jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ContainerParameters.java

```diff
@@ -195,18 +203,16 @@ public void setVolumes(List<String> volumes) {
 
   @Input
   @Optional
-  public Map<String, String> getLabels() {
-    if (System.getProperty(PropertyNames.CONTAINER_LABELS) != null) {
-      return ConfigurationPropertyValidator.parseMapProperty(
-          System.getProperty(PropertyNames.CONTAINER_LABELS));
+  public MapProperty<String, String> getLabels() {
+    String labelsProperty = System.getProperty(PropertyNames.CONTAINER_LABELS);
+    if (labelsProperty != null) {
+      Map<String, String> parsedLabels =
+          ConfigurationPropertyValidator.parseMapProperty(labelsProperty);
+      labels.set(parsedLabels);
```

Comment:
Sorry, this was missing in #3094. (I updated the issue to reflect this information.) We should call `.set()` only when the values differ. Using `.equals()` will do.
```suggestion
      if (!parsedLabels.equals(labels.get()) {
        labels.set(parsedLabels);
      }
```

Reply:
I did it this way intentionally for performance reasons, thought doing an affectation even when both maps are equal is more efficient that comparing them and do affectation only when they're different,  what do you think

Reply:
Sorry, I should have been clearer. This is not about performance but to avoid a potential error: https://github.com/GoogleContainerTools/jib/pull/3034#issuecomment-792838731

Reply:
Ah Ok my bad